esm_runscripts/esm_sim_objects.py

```diff
@@ -710,10 +750,167 @@ def tidy(self):
                 next_compute(kill_after_submit=False)
             self.end_it_all()
 
+    # NOTE(PG): Turn this into a static method once we seperate out the tidy job into plugins
+    # @staticmethod
+    def clean_run_dir(self, config):
+        """
+        This plugin allows you to clean up the ``run_${DATE}`` folder. The
+        cleanup files are specified in the config ``general`` section as
+        follows (documentation follows order of code as it is executed):
+
+        * ``general.clean_this_rundir``: (bool) Removes the entire run
+          directory.
+
+        * ``general.clean_old_rundirs_except``: (int) Removes the entire run
+          directory except for the last <x> runs
+
+        * ``general.clean_old_rundirs_keep_every``: (int) Removes the entire
+          run directory except every <x>th run
+
+        * ``general.clean_<filetype>_dir``: (bool) Erases the run directory
+          for a specific filetype
+
+        * ``general.clean_size``: (int or float) Erases all files with size
+          greater than ``clean_size``, must be specified in bytes!
+        """
+        self._clean_run_determine_user_choice(config)
+        self._clean_this_rundir(config)
+        self._clean_old_rundirs_except(config)
+        self._clean_old_runs_filetypes(config)
+        self._clean_old_runs_size(config)
+        return config
+
+
+    def _clean_run_determine_user_choice(self, config):
+        """
+        Determine user choice from a simple switch.
 
+        The user sets::
 
+        general:
+            clean_runs: <x>
 
+        where ``x`` can be one of:
 
+        * ``True`` Removes the current run dir
+        * ``False`` Keeps run dir
+        * ``int`` (must be >= 0) keep last ``x`` run dirs
+        """
+        user_clean = config["general"].get("clean_runs")
+        # TODO(PG): It might be nice if these sorts of checks happened earlier
+        # in the job, before it even gets to this function
+        if isinstance(user_clean, bool):
+            if "clean_this_run" not in config["general"]:
```

Comment:
I think this `clean_this_run` and later occurences should be changed to `clean_this_rundir` though this did not fix the error.
```
501 ------------------------------------------
502 Type Error!
503 You have set this in your config:
504 general:
505     clean_runs:  None
506 
507 This is of type:  <class 'NoneType'>
508 However, only the following types are valid:
509    * boolean
510    * integer (greater or equal to 0!)
511 Please correct that
512 ------------------------------------------
```

Reply:
Yes, also the entire thing needs to be wrapped in a `if user_clean:`. I'll push a commit in a second, if you can try again that'd be great!

Reply:
That make sense. Then the user will have 2 options?

- Option 1: simple syntax
`clean_runs` to be define as `True`, to delete all run folders, `False` to delete nothing, and a number to preserver X number of runs before the current date
- Option 2: more complex syntax where you can define the following variables: `clean_this_rundir`, `clean_old_rundirs_except`, `clean_old_rundirs_keep_every`, `clean_<filetype>_dir`, `clean_size`

Reply:
Yup exactly. The easy one should suffice for most people, but it might be that you need more granular control for whatever reason.